### PR TITLE
Update the README file and migrate OAuth2 Mock to this repository

### PR DESCRIPTION
**Pull request by barchw**
**Description**

Changes proposed in this pull request:

- Update the repository README.
- Migrate OAuth2 Mock to this repository.
- Migrate OAuth2 storage to `europe-central2-docker.pkg.dev/sap-se-cx-kyma-goat/networking-dev-tools/oauth2-mock`
- Add CODEOWNERS for Goats

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,43 +1,9 @@
...
-> Add subsections (H3) for better readability.
-
-## Usage
-
-> Explain how to use the project. You can create multiple subsections (H3). Include the instructions or provide links to the related documentation.
-
-## Development
-
-> Add instructions on how to develop the project or example. It must be clear what to do and, for example, how to trigger the tests so that other contributors know how to make their pull requests acceptable. Include the instructions or provide links to related documentation.
+This repository contains tooling, documentation and utilities that are shared between [kyma-project/istio](https://github.com/kyma-project/istio) and [kyma-project/api-gateway](https://github.com/kyma-project/api-gateway).
```
Comment (nataliasitko) [inline README.md:6]:
```suggestion
This repository contains tooling, documentation, and utilities that are shared between the projects [kyma-project/istio](https://github.com/kyma-project/istio) and [kyma-project/api-gateway](https://github.com/kyma-project/api-gateway).
```

File: README.md [hunk 2]
```diff
@@ -1,43 +1,9 @@
...
 
 ## Contributing
 <!--- mandatory section - do not change this! --->
```
Comment (nataliasitko) [inline README.md:9]: Could you also copy these files to the repo (contributing, code of conduct, licence)?

File: CODEOWNERS [hunk 3]
```diff
@@ -1,38 +1 @@
...
-# docs/ @testuser3
-# When you open a pull request that modifies the "yaml" files, only @testuser4 is requested for a review, and the global owner(s) are not.
-# *.yaml @testuser4
-
-# Reference
-
-# For more details, read the following articles on GitHub:
-# https://help.github.com/articles/about-codeowners/
-# https://github.com/blog/2392-introducing-code-owners/
+*  @kyma-project/goat
```
Comment (nataliasitko) [inline CODEOWNERS:1]:
```suggestion
*  @kyma-project/goat
*.md @kyma-project/technical-writers
```

File: oauth2-mock/README.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# OAuth2 Mock
+
+## Description
+
+This is a simple OAuth2 mock server that can be used for testing purposes. It allows you to simulate an OAuth2 server that can issue tokens and validate them.
```
Comment (nataliasitko) [inline oauth2-mock/README.md:5]:
```suggestion
This is a simple OAuth2 mock server that you can use for testing purposes. It allows you to simulate an OAuth2 server that can issue and validate tokens.
```

File: README.md (hunk 2 above)
Comment (barchw) [reply README.md:9]: https://github.com/kyma-project/networking-dev-tools/pull/1/commits/28d23ca313ad06ad21eb962082458adc4a0b5d02

**Review by triffer — APPROVED** [review]

**Pull request merged**